### PR TITLE
chore: move challenge-editor to use vite

### LINKED ISSUE #51771: migrate client challenge-editor tool to use vite setup
### Describe the Issue

 challenge-editor client  project was created using [create-react-app](https://create-react-app.dev/) tool, which by now become deprecated by react team
 thus it may be better solution to migrate this project to use [vite](https://vitejs.dev/) development configuration

### Affected Page

 challenge-editor

### Steps to Reproduce

2. Click on '...'
3. Scroll down to '...'

### Expected behavior

when running `npm run challenge-editor` script, the project should run without errors...

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by shootermv**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #51771

### CODE REVIEW

**Comment by Sboonny** [comment]
The data isn't fetched properly

image

**Comment by shootermv** [comment]
Do you check at gitpod?

**Comment by Sboonny** [comment]
yeab

**Comment by shootermv** [comment]
I think that .env file of client project should be set to url of gitpod,
instrad of localhost

**Comment by Sboonny** [comment]
cool, let me test that.

**Comment by raisedadead** [comment]
It works like a charm 🎉 !

Here is the `main` branch:

<details>
<summary>Details</summary>

image


</details>

And here is the PR:

<details>
<summary>Details</summary>

image


</details>

**Review by raisedadead — APPROVED** [review]
If you can resolve the conflicts, I will happily land this in. While the Gitpod setup is important (I haven't tested it, will follow up with debug info). It should be a separate issue because the basic local setup works as expected.

**Comment by raisedadead** [comment]
Running the main branch and the PR side by side:

image


This suggests that there is an existential issue with GitPod specifically. The console area gives more info. The warning is gone in the PR but there is an issue abou:

`body-parser deprecated undefined extended: provide extended option server.ts:45:27`

**Comment by shootermv** [comment]
thanks! lets see if  can resolve conflicts myself...
i will try and if i will give up i will ask for help

File: tools/challenge-editor/client/package.json [hunk 1]
```diff
@@ -6,33 +6,22 @@
     "@testing-library/jest-dom": "5.17.0",
     "@testing-library/react": "12.1.5",
     "@testing-library/user-event": "13.5.0",
+    "@vitejs/plugin-react": "^4.1.0",
```
Comment (Sboonny) [inline tools/challenge-editor/client/package.json:9]:
```suggestion
    "@vitejs/plugin-react": "4.1.0",
```

File: tools/challenge-editor/client/package.json [hunk 2]
```diff
@@ -6,33 +6,22 @@
...
     "codemirror": "5.65.15",
     "react": "16.14.0",
     "react-dom": "16.14.0",
     "react-router-dom": "6.16.0",
-    "react-scripts": "5.0.1",
-    "typescript": "4.9.5"
+    "typescript": "4.9.5",
+    "vite": "^4.4.9",
+    "vite-tsconfig-paths": "^4.2.1"
```
Comment (Sboonny) [inline tools/challenge-editor/client/package.json:16]:
```suggestion
    "vite": "4.4.9",
    "vite-tsconfig-paths": "4.2.1"
```

**Review by Sboonny — COMMENTED** [review]
if the issue with Gitpod is existential, then my worries aren't valid and this PR shouldn't address them, I agree with Mrugesh.

Pin the dependencies, and I am happy with this, nicely done shootermv.

**Review by Sboonny — APPROVED** [review]
LGTM, thank you shootermv.

**Comment by naomi-lgbt** [comment]
We shouldn't need the prettier type def 🤔

**Comment by shootermv** [comment]
i dont know what else can be wrong here:
image

**Review by huyenltnguyen — COMMENTED** [review]
In order to pass the CodeFactor check, I think we would need to disable the `react/no-deprecated` rule inline:

```tsx
// TODO: Renable this when we upgrade React to version 18
// eslint-disable-next-line react/no-deprecated
ReactDOM.render();
```

**Review by huyenltnguyen — APPROVED** [review]
LGTM 👍 

Thank you @shootermv!

**Pull request merged**